test/test_euler.py

```diff
@@ -0,0 +1,781 @@
+
+    p0 = 1.0
+
+    # === this next block tests 1,2,3 dimensions,
+    # with single and multiple nodes/states. The
+    # purpose of this block is to ensure that when
+    # all components of V = 0, the flux recovers
+    # the expected values (and p0 within tolerance)
+    # === with V = 0, fixed P = p0
+    tolerance = 1e-15
+    for ntestnodes in [1, 10, 100]:
+        fake_dis = MyDiscr(dim)
+        mass = cl.clrandom.rand(
+            queue, (ntestnodes,), dtype=np.float64
+        )
+        energy = cl.clrandom.rand(
+            queue, (ntestnodes,), dtype=np.float64
+        )
+        mom = make_obj_array(
+            [
+                cl.clrandom.rand(
+                    queue, (ntestnodes,), dtype=np.float64
+                )
+                for i in range(dim)
+            ]
+        )
+        p = cl.clrandom.rand(queue, (ntestnodes,), dtype=np.float64)
+
+        for i in range(ntestnodes):
+            mass[i] = 1.0 + i
+            p[i] = p0
+            for j in range(dim):
+                mom[j][i] = 0.0 * mass[i]
+        energy = p / 0.4 + 0.5 * np.dot(mom, mom) / mass
+        q = flat_obj_array(mass, energy, mom)
+        p = eos.pressure(q)
+        flux = _inviscid_flux(fake_dis, q, eos)
+
+        logger.info(f"{dim}d flux = {flux}")
+
+        # for velocity zero, these components should be == zero
+        for i in range(2 * dim):
+            for j in range(ntestnodes):
+                assert flux[i][j].get() == 0.0
+
+        # The momentum diagonal should be p
+        # Off-diagonal should be identically 0
+        for i in range(dim):
+            for j in range(dim):
+                print(f"(i,j) = ({i},{j})")
+                if i != j:
+                    for n in range(ntestnodes):
+                        assert (
+                            flux[(2 + i) * dim + j][n].get() == 0.0
+                        )
+                else:
+                    for n in range(ntestnodes):
+                        assert (
+                            flux[(2 + i) * dim + j][n].get() == p[n]
+                        )
+                        assert (
+                            np.abs(
+                                flux[(2 + i) * dim + j][n].get()
+                                - p0
+                            ) < tolerance
+                        )
+
+    # === this next block tests 1,2,3 dimensions, single and multiple nodes
+    # === with V = 1 (in some dimension), fixed P = p0
+    # In this test, we verify that there is no spurious numerical noise
+    # generated by non-zero momentum terms into unrelated flux components
+    # by checking that the terms that should be are identically zero.
+    # And we verify that the expected components have the expected values
+    # by using an input state calculated using an input pressure (P0).
+    # We expect calculated flux components to be floating point equal,
+    # but not bitwise equal to the expected fluxes.
+    for livedim in range(dim):
```

Comment:
The code below is very "wrappy" (with not much content per line) partly because of this nesting. Maybe do this loop by a pytest-parametrized test?

Reply:
Agree. I am not sure my idiotic solution is an improvement.

Reply:
Removing the `dim` loop helped. I think `livedim` can also be removed, see my suggestion above. What else don't you like about your changes?

Reply:
Ah, I think the idiotic one ended up not being right, so I backed it out.